### ISSUE TITLE
"Open profile on startup" not working (on Mac) with 2.53.3

### ISSUE THREAD

**Issue by flaschenge1st** [bcpierce00/unison#937]
In options, i can check "Open Profile on startup", but the dropdown box is empty - none of my profiles is found.

So if I start Unison there's a list of profiles from which I have to choose (as expexted if the "Open profile on startup" is not checked...)

Running 2.53.3 (ocaml 4.14.1) on a Mac M2 with Venture 13.4.

**Comment by tleedjarv** [comment]
Did this work for you before? If this is a regression then it may be related to a macOS version because uimac has not been changed in quite a while.

It would be a good to get a confirmation (from anyone) that this functionality used to work or still works for someone.

**Comment by gdt** [comment]
Please read the reporting bugs wiki page.   It's unclear how the binary you are running was built, if it's the gtk build or the mac native build.

**Comment by tleedjarv** [comment]
I don't have access to any macOS myself, so can't directly help but I can at least try help debugging the issue.

@flaschenge1st could you try a version you can download from https://github.com/tleedjarv/unison/suites/13741972879/artifacts/760772111 and see if it makes any difference?

**Comment by flaschenge1st** [comment]
Thank you! Unfortunately: Exactly the same issue.

<img width="558" alt="grafik" src="https://github.com/bcpierce00/unison/assets/37290316/e1137891-a972-4ee3-a022-bf3d29d4b282">

(the dropdown is empty!)
although there are profiles:
<img width="475" alt="grafik" src="https://github.com/bcpierce00/unison/assets/37290316/7534317f-8df3-4f2e-9eab-8011bec0783b">


Is the version you linked in your last post the gtk version? I think that's the version I had running on my old machine (the GUI of the versions is similar but not the same. The one from your link looks exactly like my old one...).

**Comment by tleedjarv** [comment]
No, this is the native mac version (it's the only one that has this functionality in the first place).

I've prepared a new debug build for you: https://github.com/tleedjarv/unison/suites/13788166276/artifacts/764443528
Could you try it and run it from the terminal? I'm expecting some debug output to be printed to the terminal. Please paste that output here.

**Comment by flaschenge1st** [comment]
```
% ./Unison.app/Contents/MacOS/Unison
2023-06-22 20:17:29.589 Unison[5005:416000] awakeFromNib called
2023-06-22 20:17:29.589 Unison[5005:416000] initProfiles called
2023-06-22 20:17:29.589 Unison[5005:416000] dir = '/Users/oschuppe/Library/Application Support/Unison'
2023-06-22 20:17:29.589 Unison[5005:416000] initProfiles count = 10
2023-06-22 20:17:39.739 Unison[5005:416000] getProfiles called
2023-06-22 20:17:39.739 Unison[5005:416000] getProfiles called
```

first 4 lines are from startup, last 2 lines come from opening the settings window.

**Comment by tleedjarv** [comment]
Thank you. It would really help to know which version of Unison you used to run. It seems that this may have been broken since all the way back starting with 2.48.3 (released in 2015(?)) or possibly since 2.51.2.

I have no experience at all with Xcode and objc but it seems to me that the code itself works fine, it's just the GUI that is broken. More specifically, the combobox seems to be disconnected from the code that should populate it. It requires someone with access to Xcode to fix this. I can't help with that.

@alanshutko (I'm just calling out blindly) perhaps you can help with this issue?

**Comment by flaschenge1st** [comment]
Oh dear - I'm sorry! My old Mac is broken, but I do have a backup! Sorry, didn't think about that!

OK, so the version I used on my old mac is

```
> ./Unison.app/Contents/MacOS/Unison -version
2023-06-27 12:54:38.449 Unison[35186:481620] Calling nonGuiStartup
unison version 2.51.4 (ocaml 4.12.0)
```

<img width="462" alt="grafik" src="https://github.com/bcpierce00/unison/assets/37290316/6bf6dd0d-4295-4e5a-b87f-571a213068d0">


Unfortunately, it shows exactly the same behaviour on the new mac. So this must be a bug in connection with the new MacOS. I found the old mac os version:

Old Mac OS version: 10.13.6 (Unison 2.51.4 is running fine)
New Mac os version: 13.4.1 (Unison 2.51.4 and 2.53.3 NOT auto-loading the profile at startup)


Thank you for your support!

**Comment by gdt** [comment]
I retitled the issue; it's of course not clear if this is a latent Unison bug or a macOS bug.  It's going to take somebody with motivation and macOS to debug it.   Intermediate reports between 10.13 and 13 are likely useful, but I suspect debugging on 13 is what is going to be necessary.

(Marked wontfix because no one who is actively maintaining uses the mac GUI.  A fix is welcome of course; wontfix is just a clue that there is no reason to expect one to appear.)

**Comment by gdt** [comment]
A reminder that this issue awaits testing by somebody who cares about macOS.

**Comment by TheMammen** [comment]
As a workaround you can set the preference from the terminal:

`defaults write  edu.upenn.cis.Unison profileToOpen Nameofprofile`

The configuration can be validated with 

`defaults read edu.upenn.cis.Unison`

**Comment by TheMammen** [comment]
I don't have time to dig through the code now, from what I can see the dropsown don't get a datasource, or an empty source.

One test could be to change the dropdown to a text field, and verify that a manuel added profile name would be written back to mac settings. This will still just be a workaround, but at least from within the UI.

The setting is read correctly from within unison, after setting it with `defaults write`
